dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -1,5 +1,8 @@
 {% macro dbt_snowflake_get_tmp_relation_type(strategy, unique_key, language) %}
-  {%- set merge_tmp_relation_type = config.get('merge_tmp_relation_type', default="view") -%}
+  {% if language == "sql"%}
+    {%- set tmp_relation_type = config.get('tmp_relation_type', default="view") -%}
```

Comment:
Is this necessary? Or can we just delete this line altogether?

In particular, I think this line will cause problems when:
```
strategy == "delete+insert" and tmp_relation_type != "table" and unique_key is not none
```

My assumption is that we can specify the subsequent logic in a way that will handle all the relevant cases if `tmp_relation_type` is not provided.

```suggestion
```

Reply:
I think that we can write the logic in such a way as to allow `tmp_relation_type` to be completely optional to the user.

In fact, we _want_ it to be completely optional rather than required.